src/frontend/src/pages/PageBase.js

```diff
@@ -25,11 +27,17 @@ const styles = {
 };
 
 function PageBase({ children, title }) {
+  const theme = useTheme();
+  const matches = useMediaQuery(theme.breakpoints.down('sm'));
   return (
     <>
       <CssBaseline />
       <SEO title={title} />
-      <Header />
+      {matches === true ? (
```

Comment:
This is leaking too much abstraction about the Header component.  Let's move all the stuff you're doing here into the Header component, and in the render function, do what you're doing here and choose which Header you actually want to display.  The `PageBase` can just use `<Header />` and you can hide all this complexity where it's needed, in the Header code itself.